compiler/rustc_session/src/config.rs

```diff
@@ -1296,6 +1296,28 @@ bitflags::bitflags! {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Sysroot {
+    pub explicit: Option<PathBuf>,
+    pub default: PathBuf,
+}
+
+impl Sysroot {
+    pub fn new(explicit: Option<PathBuf>) -> Sysroot {
+        Sysroot { explicit, default: filesearch::default_sysroot() }
+    }
+
+    /// Return explicit sysroot if it was passed with `--sysroot`, or default sysroot otherwise.
+    pub fn path(&self) -> &Path {
+        self.explicit.as_deref().unwrap_or(&self.default)
+    }
+
+    /// Returns both explicit sysroot if it was passed with `--sysroot` and the default sysroot.
+    pub fn all_paths(&self) -> impl Iterator<Item = &Path> {
```

Comment:
Maybe return `&[PathBuf]` instead? Then `fluent_bundle` won't need to be made generic.

Reply:
How? Can't do that without changing the storage in `Sysroot` to something like `SmallVec<[PathBuf; 2]>`.

`fluent_bundle` could also be made non-generic by collecting the iterator at its call site.

Reply:
Right

Reply:
I'm not sure what you meant by "Right", so I changed the `fluent_bundle` call to use `collect`, returning a slice from `all_paths` is too inconvenient.

Reply:
I meant that you are right about

> Can't do that without changing the storage in Sysroot to something like SmallVec<[PathBuf; 2]>.

I didn't think of that.

Using collect for fluent_bundle is fine.